festim/hydrogen_transport_problem.py

```diff
@@ -385,51 +396,56 @@ def assign_functions_to_species(self):
             spe.prev_solution = sub_prev_solution[idx]
             spe.test_function = sub_test_functions[idx]
 
-    def define_markers_and_measures(self):
-        """Defines the markers and measures of the model"""
+    def define_meshtags_and_measures(self):
+        """Defines the facet and volume meshtags of the model which are used
+        to define the measures fo the model, dx and ds"""
 
-        facet_indices, tags_facets = [], []
+        if isinstance(self.mesh, F.MeshFromXDMF):
+            self.facet_meshtags = self.mesh.define_surface_meshtags()
+            self.volume_meshtags = self.mesh.define_volume_meshtags()
 
-        # find all cells in domain and mark them as 0
-        num_cells = self.mesh.mesh.topology.index_map(self.mesh.vdim).size_local
-        mesh_cell_indices = np.arange(num_cells, dtype=np.int32)
-        tags_volumes = np.full(num_cells, 0, dtype=np.int32)
+        else:
+            facet_indices, tags_facets = [], []
 
-        for sub_dom in self.subdomains:
-            if isinstance(sub_dom, F.SurfaceSubdomain1D):
-                facet_index = sub_dom.locate_boundary_facet_indices(
-                    self.mesh.mesh, self.mesh.fdim
-                )
-                facet_indices.append(facet_index)
-                tags_facets.append(sub_dom.id)
-            if isinstance(sub_dom, F.VolumeSubdomain1D):
-                # find all cells in subdomain and mark them as sub_dom.id
-                self.volume_subdomains.append(sub_dom)
-                entities = sub_dom.locate_subdomain_entities(
-                    self.mesh.mesh, self.mesh.vdim
-                )
-                tags_volumes[entities] = sub_dom.id
+            # find all cells in domain and mark them as 0
+            num_cells = self.mesh.mesh.topology.index_map(self.mesh.vdim).size_local
+            mesh_cell_indices = np.arange(num_cells, dtype=np.int32)
+            tags_volumes = np.full(num_cells, 0, dtype=np.int32)
 
-        # check if all borders are defined
-        if isinstance(self.mesh, F.Mesh1D):
-            self.mesh.check_borders(self.volume_subdomains)
+            for sub_dom in self.subdomains:
```

Comment:
Linked to my comment above, I think this only applies to `Mesh1D` right?
Should we make it a method of Mesh1D?